### PR TITLE
Fixes #18264: Update Rules UI

### PR DESCRIPTION
**Pull request by RaphaelGauthier**
https://issues.rudder.io/issues/18264

### CODE REVIEW

**Comment by RaphaelGauthier** [comment]
PR updated with a new commit

**Comment by RaphaelGauthier** [comment]
PR rebased

**Comment by RaphaelGauthier** [comment]
PR updated with a new commit

File: webapp/sources/rudder/rudder-web/src/main/elm/rules/sources/DataTypes.elm [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+  | Groups
+  | TechnicalLogs
+
+type alias Targets =
+  { include : List String
+  , exclude : List String
+  }
+
+type alias RuleDetails =
+  { id                : String
```
Comment (VinceMacBuche) [inline webapp/sources/rudder/rudder-web/src/main/elm/rules/sources/DataTypes.elm:17]: You should have ID type, like RuleId, DirectiveId, and use them when needed

File: webapp/sources/rudder/rudder-web/src/main/elm/rules/sources/DataTypes.elm [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+  , policyMode       : String
+  }
+
+type alias Techniquex =
+  { name     : String
+  , versions : List String
+  }
+
+type RulesTreeItem
+  = Category String String (List RulesTreeItem) (List RulesTreeItem)
```
Comment (VinceMacBuche) [inline webapp/sources/rudder/rudder-web/src/main/elm/rules/sources/DataTypes.elm:69]: type alias RuleCategory = 
  { name : string
  , description : string
  , subcateogries : List RuleCategory
  , rules : List Rule 
  }

type RuleTreeItem = RuleCategoryItem RuleCategory | RuleItem Rule

**Comment by RaphaelGauthier** [comment]
PR updated with a new commit

**Comment by RaphaelGauthier** [comment]
PR updated with a new commit

**Comment by RaphaelGauthier** [comment]
PR rebased

**Review by VinceMacBuche — APPROVED** [review]

**Pull request merged**